src/mappings/_utils.ts

```diff
@@ -50,3 +53,44 @@ export async function isRequestExpired(
         requestHeight.active + period < latestActiveBlock
     );
 }
+
+let currencyMap = new Map<CurrencyIdentifier, CurrencyExt>();
+
+export async function currencyToLibCurrencyExt(currency: Currency): Promise<CurrencyExt> {
+    const interBtcApi = await getInterBtcApi();
+
+    let id: CurrencyIdentifier;
+    if (currency.isTypeOf === "NativeToken") {
+        id = {token: currency.token};
+    }
+    else if (currency.isTypeOf === "ForeignAsset") {
+        id = {foreignAsset: currency.asset };
+    }
+    else if (currency.isTypeOf === "LendToken") {
+        id = {lendToken: currency.lendTokenId};
+    }
+    else {
+       throw new Error("No handling implemented for currency type");
+    }
+    let currencyInfo: CurrencyExt;
+    if ( currencyMap.has(id) ) {
+        currencyInfo = currencyMap.get(id) as CurrencyExt;
+    }
+    else {
+        const currencyId = interBtcApi.api.createType("InterbtcPrimitivesCurrencyId", id );
+        currencyInfo  = await currencyIdToMonetaryCurrency(
+            interBtcApi.assetRegistry,
+            interBtcApi.loans,
+            currencyId
+        )
+
+        currencyMap.set(id , currencyInfo);
+    }
+    return currencyMap.get(id) as CurrencyExt;
+}
+
+export async function convertAmountToHuman(currency: Currency, amount: bigint ) : Promise<string> {
```

Comment:
- I prefer if we return a `bigdecimal` (https://github.com/interlay/interbtc-squid/blob/3b58a5576b3adbbbe6caf948c5706e549222b181/src/model/generated/marshal.ts#L68) type instead of stringifying the numbers.
- We should use store full precision in the float representation. `toHuman` is really a UI function as it makes some assumptions about how many decimals should be stored (and will in many cases round instead of storing all decimals)

Reply:
> * I prefer if we return a `bigdecimal` (https://github.com/interlay/interbtc-squid/blob/3b58a5576b3adbbbe6caf948c5706e549222b181/src/model/generated/marshal.ts#L68
>   ) type instead of stringifying the numbers.

Good points. 

Using a string was my suggestion yesterday as we were working through this. But I agree, returning a `bigdecimal` makes more sense.

I'm not quite sure what the corresponding Graphql schema scalar should be. I'd suggest starting with `BigDecimal` and see if it accepts it.

Reply:
Nevermind, I realized this function is used to create the custom scalar for the graphQL schema. I am working on getting bigDecimal to work with the scalar now.